dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -79,9 +79,9 @@ def update
       script = Script.get_from_cache(script_id)
       return head :bad_request if script.nil?
       # If given a course and script, make sure the script is in that course
-      return head :bad_request if course_id && course_id != script.course.try(:id)
+      return head :bad_request if course_id && course_id != script.unit_group.try(:id)
```

Comment:
Thinking about how we want this logic to look eventually is interesting. I'm hoping that we can move towards assignment always being on a CourseVersion, and then optionally it could also have a field to specify a specific part of the content for that CourseVersion. Hopefully we can build some better abstractions on that second part.

This seems reasonable for now though - we can directly assign units and unit groups for now.

Reply:
yeah, if we never assigned both UnitGroup and Unit, then this would be a great fit for just pointing to course_version. there is also the issue that assigning a unit within a unit group doesn't work very well right now, especially as some students start to move onto the next unit. some code.org employees recommend just assigning the unit_group but not the unit, and then unhiding lessons as their students start to get to them. however that assignment strategy is not what we push external users toward, and I don't think we can assume that this strategy will become the long-term plan. 

alternately, we may in the future decide to assign students to specific lessons. in that case, I would argue that we should use the **content node** abstraction idea that we discussed earlier. a `Section` would have a polymorphic association to either a unit group, unit or lesson, which implemented certain methods in the content node "interface". the content node pattern could also be used to unify the way we implement that various things are "hideable" for each section (lesson, lesson group, etc).

Since we're not really sure what our future needs are, I'll plan on the section keeping pointers to unit_group and unit. this is helpful to know, since it'll give me a bit more confidence to rename local variables, etc to match the new names.